multimap/src/main/java/org/infinispan/multimap/impl/SetBucket.java

```diff
@@ -52,8 +52,8 @@ Collection<MultimapObjectWrapper<V>> getWrappedValues() {
       return new ArrayList<>(values);
    }
 
-   public Set<V> values() {
-      return values.stream().map(MultimapObjectWrapper::get).collect(Collectors.toSet());
+   public Set<MultimapObjectWrapper<V>> values() {
+      return values;
```

Comment:
I think we need to copy here.

Reply:
I removed that method, it's not needed actually. 
I replaced it with a toList method, which returns a copy in a list